cluster/images/etcd/migrate-if-needed.sh

```diff
@@ -215,6 +215,23 @@ for step in ${SUPPORTED_VERSIONS}; do
     CURRENT_STORAGE="etcd3"
     echo "${CURRENT_VERSION}/${CURRENT_STORAGE}" > "${DATA_DIRECTORY}/${VERSION_FILE}"
   fi
+  if [ "$(echo ${CURRENT_VERSION} | cut -c1-4)" = "3.1." -a "${CURRENT_VERSION}" = "${step}" -a "${CURRENT_STORAGE}" = "etcd3" ]; then
+    # If we are upgrading to 3.1.* release, if the cluster was migrated
+    # from v2 version, the v2 data may still be around. So now is the
+    # time to actually remove them.
+    echo "Remove stale v2 data"
+    START_VERSION="${step}"
+    START_STORAGE="etcd3"
+    ETCDCTL_CMD="${ETCDCTL:-/usr/local/bin/etcdctl-${START_VERSION}}"
+    if ! start_etcd; then
+      echo "Starting etcd ${step} in v3 mode failed"
+      exit 1
+    fi
+		${ETCDCTL_CMD} rm --recursive "/registry"
```

Comment:
It looks like the `ETCDCTL_API=3 ${ETCDCTL_CMD} migrate --data-dir=${DATA_DIRECTORY}` command on line 199 should have performed an in-place migration of the data (https://coreos.com/blog/migrating-applications-etcd-v3.html#etcdctl-migrate). If we had backed up the data-dir before migrating the data, I suppose it would make sense to delete it at some point, but it doesn't look like we did that.